app/controllers/listings_controller.rb

```diff
@@ -0,0 +1,89 @@
+class ListingsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @listing = Listing.new
+    @listingList = Listing.where(user_id: current_user.id).paginate(page: params[:page], per_page: 3)
+  end
+
+  def show
+    @listing = Listing.find(params[:id])
+    @hostDetails = User.find(@listing.user_id)
+  end
+
+  def create
+    @listing = Listing.new(listing_params)
+    if @listing .save
+      flash[:notice] = 'Listing added successfully'
+      redirect_to room_listing_url(@listing)
+    else
+      flash[:alert] = 'Failed to add Listing'
+      redirect_to listings_url
+    end
+  end
+
+  def room
+    @listing = Listing.find(params[:id])
```

Comment:
You can create a private method to set the `@listing = Listing.find(params[:id])` for the methods that require it. You don't need to type the same code in every method.

Then use it in the before action.

Reference: https://guides.rubyonrails.org/action_controller_overview.html#filters